File: infra/src/main/java/mixter/infra/repositories/InMemorySessionProjectionRepository.java
package mixter.infra.repositories;

import mixter.domain.identity.SessionId;
import mixter.domain.identity.SessionProjection;
import mixter.domain.identity.SessionProjectionRepository;

import java.util.Optional;

public class InMemorySessionProjectionRepository implements SessionProjectionRepository {
    @Override
    public void save(SessionProjection sessionProjection) {

    }

    @Override
    public void replaceBy(SessionProjection sessionProjection) {

    }

    @Override
    public Optional<SessionProjection> getById(SessionId sessionId) {
        return Optional.empty();
    }


}

File: infra/src/main/scala/mixter/infra/InMemorySessionProjectionRepository.scala
package mixter.infra

import mixter.domain.identity.{SessionId, SessionProjection, SessionRepository, SessionStatus}

class InMemorySessionProjectionRepository extends SessionRepository{
  var sessions = Map.empty[SessionId,SessionProjection]

  override def save(sessionProjection: SessionProjection): Unit =
    sessions+=sessionProjection.sessionId -> sessionProjection

  override def replaceBy(sessionProjection: SessionProjection): Unit =
    save(sessionProjection)

  override def getById(id: SessionId): Option[SessionProjection] =
    sessions.get(id).filterNot(_.status==SessionStatus.DISCONNECTED)
}
